crates/bevy_scene/src/serde.rs

```diff
@@ -82,30 +90,30 @@ impl<'a> Serialize for EntitySerializer<'a> {
         let mut state = serializer.serialize_struct(ENTITY_STRUCT, 1)?;
         state.serialize_field(
             ENTITY_FIELD_COMPONENTS,
-            &ComponentsSerializer {
-                components: &self.entity.components,
+            &ReflectArraySerializer {
+                reflect_array: &self.entity.components,
                 registry: self.registry,
             },
         )?;
         state.end()
     }
 }
 
-pub struct ComponentsSerializer<'a> {
-    pub components: &'a [Box<dyn Reflect>],
+pub struct ReflectArraySerializer<'a> {
```

Comment:
Hm, what about something like `ReflectSliceSerializer` since it takes in a referenced slice? Or `ReflectMapSerializer` since it serializes to a map?

Honestly, I wonder if this should go in `bevy_reflect` (but we can consider that question out of scope for this PR haha)

Reply:
ReflectSliceSerializer sounds okay, but as you pointed out in the Deserializer struct, it could be confusing since it's pub and it doesn't really seem strictly related to scenes. I also agree it might make more sense to have this within bevy_reflect instead of bevy_scene, from my limited understanding.
The term "dynamic" is also used to refer to impl Reflect in some places within this same file, do you think something like "DynamicSliceSerializer" could be clearer? Thank you for the feedback, I'm bad at naming things haha

Reply:
Maybe we just scope this to scenes, despite the fact it could be used elsewhere. So like `SceneElementSerializer`, `SceneTypeSerializer`, `SceneMapSerializer`, `SceneItemSerializer`, or some other name. I think adding the `Scene` prefix helps differentiate it, so I’m fine with anything like that haha